// resources/procurement.js
import Authenticated from "resources/Authenticated";

export default class ProcurementResource extends Authenticated {
  id = null;
  status = "";
  supplier = null;
  staff = null;

  pk() {
    return this.id;
  }

  static declineProcurementShape() {
    return {
      ...this.updateShape(),
      getFetchKey: () => {
        return "/api/procurements/decline/";
      },
      fetch: (params: {}, body?: Readonly<object | string>) => {
        return this.fetch(
          "put",
          `${process.env.REACT_APP_API_HOST}/api/procurements/decline/${params.id}`,
          body
        );
      }
    };
  }

  static verifyProcurementShape() {
    return {
      ...this.updateShape(),
      getFetchKey: () => {
        return "/api/procurements/verify/";
      },
      fetch: (params: {}, body?: Readonly<object | string>) => {
        return this.fetch(
          "put",
          `${process.env.REACT_APP_API_HOST}/api/procurements/verify/${params.id}`,
          body
        );
      }
    };
  }

  static urlRoot = `${process.env.REACT_APP_API_HOST}/api/procurements/`;
}
