// resources/transaction.js
import Authenticated from "resources/Authenticated";

export default class TransactionResource extends Authenticated {
  id = null;
  status = "";
  customer = null;
  staff = null;
  doctor = null;
  staffId = 0;
  customerId = "";
  doctorId = 0;
  subtotal = 0;
  tax = 0;
  payAmt = 0;
  taken = false;

  pk() {
    return this.id;
  }

  static payTansactionShape() {
    return {
      ...this.updateShape(),
      getFetchKey: () => {
        return "/api/transactions/pay/";
      },
      fetch: (params: {}, body?: Readonly<object | string>) => {
        return this.fetch(
          "put",
          `${process.env.REACT_APP_API_HOST}/api/transactions/pay/${params.id}`,
          body
        );
      }
    };
  }

  static takeTransactionShape() {
    return {
      ...this.updateShape(),
      getFetchKey: () => {
        return "/api/transactions/take/";
      },
      fetch: (params: {}, body?: Readonly<object | string>) => {
        return this.fetch(
          "put",
          `${process.env.REACT_APP_API_HOST}/api/transactions/take/${params.id}`,
          body
        );
      }
    };
  }

  //   static declineProcurementShape() {
  //     return {
  //       ...this.updateShape(),
  //       getFetchKey: () => {
  //         return "/api/procurements/decline/";
  //       },
  //       fetch: (params: {}, body?: Readonly<object | string>) => {
  //         return this.fetch(
  //           "put",
  //           `${process.env.REACT_APP_API_HOST}/api/procurements/decline/${params.id}`,
  //           body
  //         );
  //       }
  //     };
  //   }

  //   static retrieveProcurementShape() {
  //     return {
  //       ...this.updateShape(),
  //       getFetchKey: () => {
  //         return "/api/procurements/retrieve/";
  //       },
  //       fetch: (params: {}, body?: Readonly<object | string>) => {
  //         return this.fetch(
  //           "put",
  //           `${process.env.REACT_APP_API_HOST}/api/procurements/retrieve/${params.id}`,
  //           body
  //         );
  //       }
  //     };
  //   }

  //   static verifyProcurementShape() {
  //     return {
  //       ...this.updateShape(),
  //       getFetchKey: () => {
  //         return "/api/procurements/verify/";
  //       },
  //       fetch: (params: {}, body?: Readonly<object | string>) => {
  //         return this.fetch(
  //           "put",
  //           `${process.env.REACT_APP_API_HOST}/api/procurements/verify/${params.id}`,
  //           body
  //         );
  //       }
  //     };
  //   }

  static urlRoot = `${process.env.REACT_APP_API_HOST}/api/transactions/`;
}
