// contexts/SidebarContext.js
import React from "react";

const sidebarItems = [
  {
    title: "Beranda",
    to: "/home",
    htmlBefore: '<i class="material-icons">home</i>',
    htmlAfter: ""
  },
  {
    title: "Karyawan",
    to: "/karyawan",
    htmlBefore: '<i class="material-icons">person</i>',
    htmlAfter: ""
  },
  {
    title: "Dokter",
    to: "/dokter",
    htmlBefore: '<i class="material-icons">local_hospital</i>',
    htmlAfter: ""
  },
  {
    title: "Supplier",
    to: "/supplier",
    htmlBefore: '<i class="material-icons">airport_shuttle</i>',
    htmlAfter: ""
  },
  {
    title: "Obat",
    to: "/obat",
    htmlBefore: '<i class="material-icons">local_hospital</i>',
    htmlAfter: ""
  },
  // {
  //   title: "Blog Posts",
  //   htmlBefore: '<i class="material-icons">vertical_split</i>',
  //   to: "/blog-posts"
  // },
  // {
  //   title: "Forms & Components",
  //   htmlBefore: '<i class="material-icons">view_module</i>',
  //   to: "/components-overview"
  // },
  // {
  //   title: "Tables",
  //   htmlBefore: '<i class="material-icons">table_chart</i>',
  //   to: "/tables"
  // },
  // {
  //   title: "User Profile",
  //   htmlBefore: '<i class="material-icons">person</i>',
  //   to: "/user-profile-lite"
  // },
  // {
  //   title: "Errors",
  //   htmlBefore: '<i class="material-icons">error</i>',
  //   to: "/errors"
  // }
];

export const defaultValue = {
  items: sidebarItems,
  active: "/home",
  visible: false,
  toggle: () => {
    console.log("hoi");
  }
};

const SidebarContext = React.createContext(defaultValue);

export default SidebarContext;
